app/assets/javascripts/course/assessment/submission/submission.js

```diff
@@ -164,14 +164,21 @@
       $('#' + identifier).find('textarea.code').ace();
     });
 
-    // Show the first tab on page load.
-    $(MULTI_QUESTION_ASSESSMENT_SELECTOR + '.tab-header a:first').tab('show');
+    var step = getParamFromURL('step=') || 1;
+    $(MULTI_QUESTION_ASSESSMENT_SELECTOR + '.tab-header a[step="' + step + '"]').tab('show');
   }
 
   function initializeAceEditor() {
     $(DOCUMENT_SELECTOR + 'textarea.code').not('.tab-content textarea.code').ace();
   }
 
+  function getParamFromURL(param) {
```

Comment:
Maybe we can go for a proper url extractor and put it into the general helper functions. This one has issues if somehow there is a `step` in any part of the URL.